src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraEntityBsonSupport.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.agora.server.model
+
+import com.mongodb.DBObject
+import org.bson.types.ObjectId
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+
+
+object AgoraEntityBsonSupport {
+
+  def read(obj: DBObject): AgoraEntity =
+    AgoraEntity(
+      namespace       = stringOrNone(obj, "namespace"),
+      name            = stringOrNone(obj, "name"),
+      snapshotId      = intOrNone(obj, "snapshotId"),
+      snapshotComment = stringOrNone(obj, "snapshotComment"),
+      synopsis        = stringOrNone(obj, "synopsis"),
+      documentation   = stringOrNone(obj, "documentation"),
+      owner           = stringOrNone(obj, "owner"), // Not in DB but checked by tests. Just rolling with this for now.
+      createDate      = dateOrNone(obj, "createDate"),
+      payload         = stringOrNone(obj, "payload"),
+      payloadObject   = None, // Not in DB
+      url             = None, // Not in DB
+      entityType      = entityType(obj),
+      id              = Option(new ObjectId(obj.get("_id").toString)),
+      methodId        = idOrNone(obj, "methodId"), // Configs only
```

Comment:
Use `idOrNone` for `id` just like `methodId`.

`id` _should not_ come back as `null`, but `AgoraEntity` previously allowed it to be `null`, so I say Keep It Short & Simple and let `id` still possibly be `null`/`None` during your refactored parsing.

The full block using type inference:

```scala
AgoraEntity(
  namespace = getOrNone(obj, "namespace"),
  name = getOrNone(obj, "name"),
  snapshotId = getOrNone(obj, "snapshotId"),
  snapshotComment = getOrNone(obj, "snapshotComment"),
  synopsis = getOrNone(obj, "synopsis"),
  documentation = getOrNone(obj, "documentation"),
  owner = getOrNone(obj, "owner"), // Not in DB but checked by tests. Just rolling with this for now.
  createDate = dateOrNone(obj, "createDate"),
  payload = getOrNone(obj, "payload"),
  payloadObject = None, // Not in DB
  url = None, // Not in DB
  entityType = entityType(obj),
  id = getOrNone(obj, "_id"),
  methodId = getOrNone(obj, "methodId"), // Configs only
  method = None, // Not in DB
  managers = Seq.empty, // Not in DB
  public = None, // Not in DB
)
```